src/Jobs/IndexJob.php

```diff
@@ -26,6 +28,11 @@ class IndexJob extends BatchJob
 
     use Injectable;
     use Extensible;
+    use ConfigurationAware;
+
+    private static array $dependencies = [
+        'configuration' => '%$' . IndexConfiguration::class,
+    ];
```

Comment:
I've been wondering a little bit about how valuable including these as dependencies is.

* The dependencies are not available in the `__construct()` (which is annoying)
* We can just access them using `class::singleton()` anyway